app/models/design_factor.rb

```diff
@@ -1,6 +1,13 @@
 class DesignFactor < ActiveRecord::Base
+  belongs_to :user
+
+  after_update { plant_scoring_units.each(&:touch) }
+  before_destroy { plant_scoring_units.each(&:touch) }
+
   has_many :plant_scoring_units
 
+  validates :design_factors, presence: true
+
   validates :design_factor_name,
```

Comment:
An old CS column. Not really useful in this context. In CS times they used names as FKeys (not even real FKeys, just values), so they needed (sometimes artificial) name column everywhere.